src/core/connection_pool.c

```diff
@@ -0,0 +1,490 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+Abstract:
+
+    Definitions for the MsQuic Connection Pool API, which allows clients to
+    create a pool of connections that are spread across RSS cores.
+
+--*/
+
+#include "precomp.h"
+
+#ifdef QUIC_CLOG
+#include "connection_pool.c.clog.h"
+#endif
+
+#define MAX_CONNECTION_POOL_RETRY_MULTIPLIER 2
+
+_IRQL_requires_max_(PASSIVE_LEVEL)
+QUIC_STATUS
+QuicConnPoolGetStartingLocalAddress(
+    _In_ QUIC_ADDR* RemoteAddress,
+    _Out_ QUIC_ADDR* LocalAddress
+    )
+{
+    CXPLAT_SOCKET* Socket = NULL;
+    CXPLAT_UDP_CONFIG UdpConfig = { .RemoteAddress = RemoteAddress, };
+    QUIC_STATUS Status =
+        CxPlatSocketCreateUdp(MsQuicLib.Datapath, &UdpConfig, &Socket);
+    if (QUIC_FAILED(Status)) {
+        goto Error;
+    }
+
+    CxPlatSocketGetLocalAddress(Socket, LocalAddress);
+
+Error:
+    if (Socket != NULL) {
+        CxPlatSocketDelete(Socket);
+    }
```

Comment:
```suggestion
    QUIC_STATUS Status =
        CxPlatSocketCreateUdp(MsQuicLib.Datapath, &UdpConfig, &Socket);
    if (QUIC_SUCCEEDED(Status)) {
        CXPLAT_DBG_ASSERT(Socket != NULL);
        CxPlatSocketGetLocalAddress(Socket, LocalAddress);
        CxPlatSocketDelete(Socket);
    }
```
This change eliminates the 'you don't have code coverage' warning for the error case. Now there is no error case 😄